core/src/main/scala/org/apache/spark/metrics/MetricsConfig.scala

```diff
@@ -57,14 +59,16 @@ private[spark] class MetricsConfig(conf: SparkConf) extends Logging {
       case _ =>
     }
 
-    propertyCategories = subProperties(properties, INSTANCE_REGEX)
-    if (propertyCategories.contains(DEFAULT_PREFIX)) {
-      val defaultProperty = propertyCategories(DEFAULT_PREFIX).asScala
-      for((inst, prop) <- propertyCategories if (inst != DEFAULT_PREFIX);
+    perInstanceProperties = subProperties(properties, INSTANCE_REGEX)
+    if (perInstanceProperties.contains(DEFAULT_PREFIX)) {
+      val defaultProperty = perInstanceProperties(DEFAULT_PREFIX).asScala
+      for((inst, prop) <- perInstanceProperties if (inst != DEFAULT_PREFIX);
           (k, v) <- defaultProperty if (prop.get(k) == null)) {
         prop.put(k, v)
       }
     }
+
+    metricsNamespaceConfParam = properties.getProperty("namespace-conf-param", "spark.app.id")
```

Comment:
I'm not a fan of this config name. How about one of:

- metrics.namespace
- metrics.namePrefix
- metrics.appName

Or something like those?